### ISSUE TITLE
SEPA Direct Debit not working properly

### ISSUE THREAD

**Issue by flamboy** [wirecard/magento2-ee#370]
### Prerequisites

- [x] I have my [server set up correctly](https://github.com/wirecard/magento2-ee/wiki/Installation#requirements).
- [x] I am using a [supported version of Magento](https://github.com/wirecard/magento2-ee/wiki#overview-magento-2-shop-plugin).

### Versions

**Magento version:** 2.3.2-p2
**PHP version:** 7.2
**Plugin name:** Wirecard_ElasticEngine
**Plugin version:** 3.0.1

### Issue

**Problem description:**

An issue found in SEPA Direct debit
Can't purchase any product.
Magento set to developer mode.
Also tried with test/live credentials.

**Steps to reproduce:**

1. Set-up & activate payment method
2. Clean-up static content & caches
3. Go to checkout page
4. Choose _Wirecard SEPA Direct Debit_ & click _Place order_
5. Customer **can't accept** order, also an error in console appeared
https://i.imgur.com/qE7lABr.png

**Temporary fix**
1. Re-write _https://github.com/wirecard/magento2-ee/blob/v3.0.1/view/frontend/web/js/view/payment/method-renderer/sepa.js#L84_ at theme design with custom URL
2. Created own action with execute method with **layout** return type
```
public function execute()
    {
        return $this->resultFactory->create(ResultFactory::TYPE_LAYOUT);
    }
```
3. Create own layout with exist sepa-block, instead whole page creation & broken blocks https://i.imgur.com/94ujxwK.png
```
<?xml version="1.0"?>
<layout xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:framework:View/Layout/etc/layout_generic.xsd">
    <container name="root">
        <block class="Wirecard\ElasticEngine\Block\Checkout\SepaMandateBlock" template="Wirecard_ElasticEngine::sepamandate.phtml" name="frontend.sepamandate" />
    </container>
</layout>
```

**Comment by rinnhofer** [comment]
Hello @flamboy ,

thank you for the detailed contribution and your solution proposal. We will look into it as soon as possible and give you feedback when we are working on a solution.

Kind regards,
Wirecard Shop Systems Team

**Comment by vitali-graz** [comment]
Hello @flamboy,

we have looked into the issue and it is fixed in our latest release.
Please try updating to [version 3.1.0](https://github.com/wirecard/magento2-ee/releases/tag/v3.1.0) and see if you are still experiencing the issue there.

Best regards,
Wirecard Shop Systems Team

**Comment by flamboy** [comment]
So, in which place it was fixed (3.1.0) ? 
I still got the same errors https://i.imgur.com/Wb88SWt.png

**Comment by vitali-graz** [comment]
@flamboy 
We've tested it with 3.0.1 and 3.1.0 (on magento 2.3.3) and were not able to reproduce the error.

Can you please contact our support directly at shop-systems-support@wirecard.com with a description of your issue? I'm sure, our support will be able to help you pinpoint the problem root cause.

Best regards,
Wirecard Shop Systems Team

**Comment by flamboy** [comment]
@vitali-graz damn, guys. You are uses backend blocks on frontend & no issues. really?

**Comment by vitali-graz** [comment]
@flamboy you have a valid point there.
It doesn't throw errors in our setup but we should definitely improve this part. Thanks for your feedback, we will look into this in the near future.
Stay tuned to upcoming releases.

Best regards,
Wirecard Shop Systems Team

**Comment by tomazpu** [comment]
Hello @flamboy,

today I opened a pull request with the fix to the problem. It will be released in the next release.

Best regards,
Tomaz
 Wirecard Shop Systems Team

**Comment by tomazpu** [comment]
Hello @flamboy, 

we released the version 3.1.2 in which this issue is fixed. Thank you for the report.

Best Regards,
Tomaz
